File: 2394-count-subarrays-with-score-less-than-k/2394-count-subarrays-with-score-less-than-k.py
class Solution:
    def countSubarrays (self, nums : List [int], k : int) -> int:
        count = 0
        sum = 0
        left = 0
        for right in range (len (nums)):
            sum += nums [right]
            while sum * (right - left + 1) >= k:
                sum -= nums [left]
                left += 1
            count += (right - left + 1)
        return count

File: 2394-count-subarrays-with-score-less-than-k/2394-count-subarrays-with-score-less-than-k.c
long long countSubarrays (int* nums, int numsSize, long long k) {
    long long count = 0, sum = 0;
    int left = 0;
    for (int right = 0; right < numsSize; right++) {
        sum += nums [right];
        while (sum * (right - left + 1) >= k)
            sum -= nums [left++];
        count += (right - left + 1);
    }
    return count;
}